src/components/home/index.js

```diff
@@ -1,136 +1,72 @@
-    this.btn = this.btn.bind(this);
-  }
-
-  handleClick() {
-    this.setState(some => ({
-      isToggleOn: !some.isToggleOn
-    }));
-  }
-
-  btn(id) {
-    this.setState({
-      todos: this.state.todos.filter(item => item.id !== id)
-    });
-  }
-
   state = {
     todos: [],
-    editPopup: null
+    editPopup: null,
+    textEdit: []
   };
 
-  handleAdd = (title, car) => {
-    const state = this.state.todos;
-    const arr = {
-      id: state.length + 1,
-      title: title,
-      desc: car
-    };
-    state.push(arr);
-    this.setState({
-      todos: state
-    });
+  addTask = (title, car) => {
+    const {addTextTask} = this.props;
+    addTextTask(title, car)
   };
 
-  componentDidMount = () => {
-    setTimeout(() => {
-      this.setState({
-        todos: [
-          {
-            id: 1,
-            title: "Car",
-            desc: "Subaru"
-          },
-          {
-            id: 2,
-            title: "Car",
-            desc: "Audi"
-          },
-          {
-            id: 3,
-            title: "Car",
-            desc: "Mercedes"
-          },
-          {
-            id: 4,
-            title: "Car",
-            desc: "BMW"
-          }
-        ]
-      });
-    }, 0);
+  handleRemoveClick = id => {
+    const {props: { removeTextTask } } = this;
+    removeTextTask(id);
   };
 
-  edit = data => {
-    const {state:{todos}} = this;
-    const editedArray = todos.map(item => {
-      if (item.id === data.id) {
-        item.title = data.title;
-        item.desc = data.desc;
-        console.log(item.title + item.desc)
-        return item;
-      }
-      return item;
-    });
-    
-    this.setState({todos: editedArray});
+
+  editTask = data => {
+    const {editTextTask} = this.props;
+    editTextTask(data)
     this.closeEditing();
-  }
+  };
 
   openEditing = id => {
-    const {
-      state: {
-        todos,
-      }
-    } = this
+    const {todos} = this.props
+
     if(todos.length > 0){
-        const edittableItem = todos.find(item => {
-          return item.id === id
-        })
-        if(edittableItem){
-          this.setState({editPopup: (
-            <Modal id={edittableItem.id} edit={this.edit} title={edittableItem.title} desc={edittableItem.desc}/>
-          )
-        })
-       }
+      const todo = todos.find(item => {
+        return item.id === id
+      })
+      if(todo){
+        this.setState({textEdit: todo})
+        this.setState({editPopup: true})
+      }
     }
-  }
+  };
+  
   closeEditing = () => {
     this.setState({editPopup: null})
-  }
-
+  };
+  
   render() {
-    const {
-      state: {
-        todos,
-        editPopup
-      }
-    } = this
+    const {todos} = this.props
+    const {editPopup, textEdit} = this.state;
+    
     return (
       <div className="App">
         <Header />
-        {!!editPopup && editPopup}
+        {editPopup && <Modal edit={this.editTask} id={textEdit.id} title={textEdit.title} desc={textEdit.desc}/>}
```

Comment:
`<Modal onEdit={this.handleEditTask} id={textEdit.id} todo={this.state.selectedTodo} />`